lib/langchain/llm/openai.rb

```diff
@@ -115,20 +117,27 @@ def chat(prompt: "", messages: [], context: "", examples: [], **options)
 
       parameters = compose_parameters @defaults[:chat_completion_model_name], options
       parameters[:messages] = compose_chat_messages(prompt: prompt, messages: messages, context: context, examples: examples)
-      parameters[:max_tokens] = validate_max_tokens(parameters[:messages], parameters[:model])
+
+      unless functions
+        parameters[:max_tokens] = validate_max_tokens(parameters[:messages], parameters[:model])
+      end
+
+      unless functions.nil? || functions.empty?
+        parameters[:functions] = functions
+      end
```

Comment:
I think this can be simplified with:
```ruby
if functions
  parameters[:functions] = functions
else
  parameters[:max_tokens] = validate_max_tokens(parameters[:messages], parameters[:model])
end
```